examples/bits.rs

```diff
@@ -1,98 +1,207 @@
+    // Place the turtle according to the length of the text span
+    let top_edge = BIT_HEIGHT * (TEXT.len() as f64 / 2.0);
+    // The turtle starts from the top right of the region,
+    turtle.go_to((right_edge, top_edge));
+    // and walks left
+    turtle.set_heading(180.0);
 
-/// The number of bits in a single element of the bytes Vec in BitVec
-/// Set to 8 because BitVec stores u8 values. If we stored u16, we would use 16 here
-const BITS: usize = 8;
-
-/// An implementation of a vector of bits that can be easily set individually
-/// Note that this is meant to be a teaching implementation and so is not necessarily the most
-/// sophisticated or performant way to do everything. The goal is to make the code understandable
-/// and correct.
-/// See the bit-vec crate and others on crates.io for more advanced implementations.
-#[derive(Debug, Clone, Default)]
-pub struct BitVec {
-    bytes: Vec<u8>,
-}
+    // Render the text as bits of memory.
 
-impl fmt::Display for BitVec {
-    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        // EXERCISE: Only print the bytes up to the length of bytes that have been pushed, not
-        // the entire capacity. (Similar to how Vec prints its len, not its capacity)
-        for byte in &self.bytes {
-            write!(f, "{:0width$b}", byte, width = BITS)?;
-        }
-        Ok(())
-    }
-}
+    // Rust strings can iterate over their individual characters. This block
+    // loops over characters, collecting their start point in the text so that
+    // we can grab the encoded bytes of each one.
+    let mut row_num = 0;
+    for (char_num, (start, codepoint)) in TEXT.char_indices().enumerate() {
+        println!("Character {}: {}", char_num, codepoint);
+        // Each character has a variable width, so we need to find that.
+        let byte_count = codepoint.len_utf8();
+        // And then collect the bytes of the string that make up the character.
+        // `start` gives us the starting position in the text sequence, and
+        // `byte_count` gives us the length in bytes of the character, so we
+        // need to select the range beginning at `start`, running for
+        // `byte_count`. Another style of writing this that you might see in
+        // Rust libraries is `[start ..][.. length]`.
+        let row: &[u8] = &TEXT.as_bytes()[start .. start + byte_count];
+
+        // For each byte (`u8`), we use `bitvec` to make a view into its bits.
+        // `bitvec` provides the `.bits::<_>()` method on Rust integers for easy
+        // access to its view types.
+        //
+        // The `LittleEndian` means that the view moves from least significant
+        // bit to most significant. Since we want to display on screen the most
+        // significant bit on the left, and the least on the right, the turtle
+        // will have to move from right to left to match.
+        //
+        // These types describe different ways to view the same data. You can
```

Comment:
"These types" should probably be replaced with "The `LittleEndian` and `BigEndian` types". The sentence made sense before, but it's fallen out of context with the recent changes so we need to be a little more specific. :)